src/modules/fancyzones/lib/JsonHelpers.cpp

```diff
@@ -1101,222 +406,211 @@ namespace JSONHelpers
+		if (std::filesystem::exists(tmpFilePath))
         {
-            CustomZoneSetJSON result;
-
-            result.uuid = customZoneSet.GetNamedString(L"uuid");
-            if (!isValidGuid(result.uuid))
+            if (auto zoneSetJson = json::from_file(tmpFilePath); zoneSetJson.has_value())
             {
-                return std::nullopt;
+                if (auto deviceInfo = JSONHelpers::DeviceInfoJSON::FromJson(zoneSetJson.value()); deviceInfo.has_value())
+                {
+                    result = std::move(deviceInfo);
+                }
             }
+        }
+        // TODO(stefan): Check why this was inside triple if above
+		DeleteTmpFile(tmpFilePath);
+        return result;
+    }
 
-            result.data.name = customZoneSet.GetNamedString(L"name");
-
-            json::JsonObject infoJson = customZoneSet.GetNamedObject(L"info");
-            std::wstring zoneSetType = std::wstring{ customZoneSet.GetNamedString(L"type") };
-            if (zoneSetType.compare(L"canvas") == 0)
+    std::optional<CustomZoneSetJSON> ParseCustomZoneSetFromTmpFile(std::wstring_view tmpFilePath)
+    {
+        std::optional<CustomZoneSetJSON> result{ std::nullopt };
+        if (std::filesystem::exists(tmpFilePath))
+        {
+            try
             {
-                if (auto info = CanvasLayoutInfo::FromJson(infoJson); info.has_value())
-                {
-                    result.data.type = CustomLayoutType::Canvas;
-                    result.data.info = std::move(info.value());
-                }
-                else
+                if (auto customZoneSetJson = json::from_file(tmpFilePath); customZoneSetJson.has_value())
                 {
-                    return std::nullopt;
+                    if (auto customZoneSet = JSONHelpers::CustomZoneSetJSON::FromJson(customZoneSetJson.value()); customZoneSet.has_value())
+                    {
+                        result = std::move(customZoneSet);
+                    }
                 }
             }
-            else if (zoneSetType.compare(L"grid") == 0)
+            catch (const winrt::hresult_error&)
             {
-                if (auto info = GridLayoutInfo::FromJson(infoJson); info.has_value())
-                {
-                    result.data.type = CustomLayoutType::Grid;
-                    result.data.info = std::move(info.value());
-                }
-                else
+                result = std::nullopt;
+            }
+
+            DeleteTmpFile(tmpFilePath);
+        }
+        return result;
+    }
+
+    std::vector<std::wstring> ParseDeletedCustomZoneSetsFromTmpFile(std::wstring_view tmpFilePath)
+    {
+        std::vector<std::wstring> result{};
+        if (std::filesystem::exists(tmpFilePath))
+        {
+            auto deletedZoneSetsJson = json::from_file(tmpFilePath);
+            try
+            {
+                auto deletedCustomZoneSets = deletedZoneSetsJson->GetNamedArray(L"deleted-custom-zone-sets");
```

Comment:
Replace all strings literals with const strings and explicitly mark them as non-localizable.